api/proto/teleport/legacy/types/types.proto

```diff
@@ -1580,6 +1582,44 @@ message ProvisionTokenSpecV2GCP {
   repeated Rule Allow = 1 [(gogoproto.jsontag) = "allow,omitempty"];
 }
 
+// ProvisionTokenSpecV2Terraform contains Terraform-specific parts of the
+// ProvisionTokenSpecV2.
+message ProvisionTokenSpecV2Terraform {
+  // Rule is a set of properties the Terraform-issued token might have to be
+  // allowed to use this ProvisionToken.
+  message Rule {
```

Comment:
Is every field required? Or is there any optional one?

Reply:
Technically they are all optional with some verification rules (some org name/id + at least one other parameter). I'll make sure the docstring here mentions it.